src/pudl/analysis/allocate_net_gen.py

```diff
@@ -860,20 +851,19 @@ def identify_proposed_plants(gen_assoc):
     return proposed_plants
 
 
-def _allocate_unassociated_records(
+def _allocate_unassociated_bf_records(
     gen_assoc: pd.DataFrame,
     idx_cols: list[str],
     col_w_unexpected_codes: Literal["energy_source_code", "prime_mover_code"],
     data_columns: list[str],
 ) -> pd.DataFrame:
-    """Associate unassociated gen_fuel table records on idx_cols.
+    """Associate unassociated :ref:`boiler_fuel_eia923` table records on idx_cols.
```

Comment:
several aspects of this function were designed for either the "energy_source_code" or the "prime_mover_code" being unassociated. But upon further reflection and in-notebook dissecting, we were using this to merge in the unassocaited records from the gf table and then the bf table. I converted it to be a little more tailored to the bf table.